stellargraph/core/graph.py

```diff
@@ -196,41 +196,59 @@ def has_node(self, node: Any) -> bool:
         """
         raise NotImplementedError
 
-    def neighbors(self, node: Any) -> Iterable[Any]:
+    def neighbors(
+        self, node: Any, include_edge_weight=False, edge_types=None
+    ) -> Iterable[Any]:
         """
         Obtains the collection of neighbouring nodes connected
         to the given node.
 
         Args:
             node (any): The node in question.
+            include_edge_weight (bool, default False): If True, each neighbour in the
+                output is a tuple of (node ID, edge weight)
+            edge_types (list of hashable, optional): If provided, only traverse the graph
+                via the provided edge types when collecting neighbours.
 
         Returns:
             iterable: The neighbouring nodes.
         """
         raise NotImplementedError
 
-    def in_nodes(self, node: Any) -> Iterable[Any]:
+    def in_nodes(
+        self, node: Any, include_edge_weight=False, edge_types=None
+    ) -> Iterable[Any]:
         """
         Obtains the collection of neighbouring nodes with edges
         directed to the given node. For an undirected graph,
         neighbours are treated as both in-nodes and out-nodes.
 
         Args:
             node (any): The node in question.
+            include_edge_weight (bool, default False): If True, each neighbour in the
+                output is a tuple of (node ID, edge weight)
```

Comment:
With the named tuple, this could be

```suggestion
                output is a named tuple with fields `node` (the node ID) and `weight` (the edge weight)
```

or similar.